clarity_ext/extensions.py

```diff
@@ -73,164 +77,180 @@ def _run_path(self, args, module, mode, config):
             # so they don't get mixed up:
             module_parts = module.split(".")[1:]
             path = os.path.sep.join(module_parts)
-            return os.path.join(root, path, args["pid"], "run-" + mode)
+            return os.path.join(root, path, pid, "run-" + mode)
         else:
             raise ValueError("Unexpected mode")
 
-    def _parse_run_argument(self, in_argument):
-        if isinstance(in_argument, IntegrationTest):
-            return in_argument.run_argument_dict
-        if isinstance(in_argument, str):
-            return {"pid": in_argument}
-        elif isinstance(in_argument, dict):
-            return in_argument
-        else:
-            return in_argument.__dict__
-
     def _artifact_service(self, pid):
         session = ClaritySession.create(pid)
         step_repo = StepRepository(session, DEFAULT_UDF_MAP)
         return ArtifactService(step_repo)
 
-    def execute(self, module, mode, run_arguments_list=None, config=None, artifacts_to_stdout=True,
-                print_help=True, use_cache=None):
+    def run_exec(self, config, run_arguments_list, module):
+        """Executes the extension normally, without freezing or caching. This should be the default in production."""
+        for run_arguments in run_arguments_list:
+            pid = run_arguments["pid"]
+            path = self._get_run_path(pid, module, self.RUN_MODE_EXEC, config)
+            self._run(path, pid, module, False, False)
+
+    def run_test(self, config, run_arguments_list, module, artifacts_to_stdout, use_cache):
+        self.msg("To execute from Clarity:")
+        self.msg("  clarity-ext extension --args '{}' {} {}".format(
+            "pid={processLuid}",
+            module, self.RUN_MODE_EXEC))
+        self.msg("To freeze the latest test run (set as reference data for future validations):")
+        self.msg("  clarity-ext extension {} {}".format(
+            module, self.RUN_MODE_FREEZE))
+
+        if use_cache is None:
+            use_cache = True
+        self._set_cache(use_cache)
+        instance = self._get_extension(module)(None)
+
+        self._prepare_runs(instance)  # Required to support certain tests
+        if not run_arguments_list:
+            run_arguments_list = self._gather_runs(module, True)
+
+        for run_arguments in run_arguments_list:
+            pid = run_arguments["pid"]
```

Comment:
Perhaps it just shows my ignorance for the arguments, but shouldn't you validate that these steps exist? If no `run_arguments_list` is passed into `run_test()` then `NoTestsFoundException` seem to be raised inside `_gather_runs()`, but if the user has supplied invalid `args` on the CLI level then I guess it would fail somewhere else, further down the rabbit hole (perhaps in the `ClaritySession` layer, but it is not immediately clear to me)?